api/envoy/config/rbac/v3/rbac.proto

```diff
@@ -358,6 +362,9 @@ message Principal {
     // Identifies the principal using a filter state object.
     type.matcher.v3.FilterStateMatcher filter_state = 12;
 
+    // Glob URL path matching.
+    envoy.extensions.path.match.uri_template.v3.UriTemplateMatchConfig glob_path = 13;
```

Comment:
@envoyproxy/api-shepherds I'm wondering if type.matcher.v3.PathMatcher should absorb envoy.extensions.path.match.uri_template.v3.UriTemplateMatchConfig somehow?

The problem is this is core depending on some extension. But, maybe we want to consider promoting URI template matching to core? Not sure, but when I see a change like this it seems worth asking.

Reply:
I.e. allow URI template matcher to be plugged in. In any case, we should not be referencing the extension config directly here and instead should have something like https://github.com/envoyproxy/envoy/blob/b039d8a316a081d1eae3e3222e045b749d26765e/api/envoy/config/route/v3/route_components.proto#L601 if we don't end up putting it inside PathMatcher.

Reply:
hey, I'm going on holiday so I will rework it once I'm back, probably next year